### ISSUE TITLE
Select item of secondary drawer

### ISSUE THREAD

**Issue by tommyd3mdi** [mikepenz/MaterialDrawer#1239]
I have two drawers, one left and one right-mounted one. Now I want to pre-select an item - which is a sub-item of an `ExpandableDrawerItem` - on the right one and did so while creating the item itself (`.withSetSelected(true)`).

This works all fine, until I collapse and expand the top item again - this looses the selected item. Ok, fine I thought, surely the click on the top item makes it somehow selected, so let's re-select the original item after the click happened. However, I cannot see the adapter that actually holds my secondary drawer's items, I only see the `ItemAdapter` of the primary (left) one.

I guess this is conceptionally not possible, because `DrawerBuilder`s `append(Drawer)` method only takes the `DrawerLayout` from the other drawer but not keeps any of the other Drawer-related things around, like the `DrawerBuilder` that contains the reference to the other `ItemAdapter`, right?

**Comment by mikepenz** [comment]
@tommyd3mdi hi. Ok your question / issue has multiple steps and also some possible multiple solutions :D

Coming to the first issue that selections within an `Expandable` group are dismissed when collapsing. This is currently a limitation of the `FastAdapter` I use here. I am currently testing a new feature which will allow exactly your usecase, and you can read about it here:
https://github.com/mikepenz/FastAdapter/issues/122 
(This might not yet completely solve the topic for the MaterialDrawer as some functions are still position depending, and an item which is selected inside a collapsed item has no position ;))

Coming to your second issue, that setting `withSetSelected(true)` does not actually select your item. Just by changing the data of an item would not change the ui as the `RecyclerView` holding the item does not yet know about the fact that it was changed. Normally you would just call `notifyItemChanged()` on your adapter. As this is already handled by the `FastAdapter` and so also by the `MaterialDrawer` your solutions here is that you just select the item via the `Drawer` object. (which brings is to your third issue :D)

So you are right, the `.append()` will use the `DrawerLayout` from the primary drawer. But will still return a `Drawer` object related to the Appended-`Drawer` so you remember both `Drawer` objects, from your first and, appended (second) drawer and will be able to handle everything through those objects. So `drawer.getRecyclerView()` will give you the RV of the main drawer, and `appended.getRecyclerView()` will give you the RV from the appended drawer. ;). You can also use those methods to select / deselect, notify about changed items as written in the prior section. 

I hope that helps ;)

**Comment by tommyd3mdi** [comment]
@mikepenz Yep, actually I solved it this way - I remember both `Drawer` objects now and select the items in a distinct manner (which makes the most sense). Maybe I missed the documentation on this part, but it was not so totally obvious to me that I had to do that. I rather expected that since I appended one object on another, I could still somehow get the reference out of it.

I also use with `Drawer.setSelection(identifier, false)` now to select items after I build the menu and everytime the expandable menu item is clicked and this works great.

Btw... very nice API design you have there, really well thought-through!
